rust-webvr/src/api/vrexternal/mod.rs

```diff
@@ -0,0 +1,83 @@
+mod display;
+mod service;
+
+use libc;
+use std::os::raw::c_void;
+use {VRService, VRServiceCreator};
+
+#[allow(non_upper_case_globals)]
+#[allow(non_camel_case_types)]
+#[allow(non_snake_case)]
+#[allow(dead_code)]
+mod mozgfx {
+    include!(concat!(env!("OUT_DIR"), "/moz_external_vr.rs"));
+}
+
+#[derive(Clone)]
+pub struct VRExternalShmemPtr(*mut mozgfx::VRExternalShmem);
+
+unsafe impl Send for VRExternalShmemPtr {}
+unsafe impl Sync for VRExternalShmemPtr {}
+
+impl VRExternalShmemPtr {
+    pub fn new(raw: *mut c_void) -> VRExternalShmemPtr {
+        VRExternalShmemPtr(raw as *mut mozgfx::VRExternalShmem)
+    }
+
+    pub fn pull_system(&self, exit_cond: &Fn(&mozgfx::VRSystemState) -> bool) -> mozgfx::VRSystemState {
+        unsafe {
+            let mutex = &mut (*self.0).systemMutex as *mut _ as *mut libc::pthread_mutex_t;
```

Comment:
maybe create some util method to get the mutexes and condition variables easier? it's repeated in different functions and it's not a easy cast